controllers/bsl.go

```diff
@@ -93,11 +93,7 @@ func (r *DPAReconciler) ValidateBackupStorageLocations(dpa oadpv1alpha1.DataProt
 	return true, nil
 }
 
-func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger) (bool, error) {
-	dpa := oadpv1alpha1.DataProtectionApplication{}
-	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
-		return false, err
-	}
+func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger, dpa oadpv1alpha1.DataProtectionApplication) (bool, error) {
```

Comment:
I would make this a pointer, otherwise the param being passed in will be a copy, and if there's any changes, it might not persist to the next functions.
```suggestion
func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger, dpa *oadpv1alpha1.DataProtectionApplication) (bool, error) {
```

Reply:
i agree, was thinking of the same.

Reply:
fixed at all places.